### PR TITLE
fix: import Talos with initramfs generation fixes

### PR DESCRIPTION
**Pull request by smira**
* fix depmod for non-current kernel versions
* fix early u-code building

Add tests for initramfs contents.

### CODE REVIEW

File: internal/integration/initramfs_test.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+			return
+		}
+	}
+}
+
+func assertInitramfs(t *testing.T, path string, expected initramfsSpec) {
+	t.Helper()
+
+	f, err := os.Open(path)
```
Comment (frezbo) [inline internal/integration/initramfs_test.go:58]: not closed?

File: internal/integration/initramfs_test.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+			}
+
+			actualEarlyPaths = append(actualEarlyPaths, record.Name)
+		}
+
+		for _, expectedPath := range expected.earlyPaths {
+			assert.Contains(t, actualEarlyPaths, expectedPath)
+		}
+
+		eatPadding(t, in)
```
Comment (frezbo) [inline internal/integration/initramfs_test.go:90]: ahh so there's padding indeed at start of each section?

File: internal/integration/initramfs_test.go (hunk 1 above)
Comment (smira) [reply internal/integration/initramfs_test.go:58]: it's closed at the very bottom of this function

File: internal/integration/initramfs_test.go (hunk 2 above)
Comment (smira) [reply internal/integration/initramfs_test.go:90]: I think `cpio` adds it, even though I can't find a word about it, but kernel supports it, so we do same

**Review by frezbo — APPROVED** [review]

**Pull request merged**